lib/mongo/server/connection_pool/connection_pool_populator.rb

```diff
@@ -39,9 +43,10 @@ def do_work
           unless @pool.populate
             @pool.populate_semaphore.wait
           end
-        rescue Exception
+        rescue Error => e
           # Errors encountered when trying to add connections to
           # pool; try again later
+          log_warn("Populator failed to connect a connection due to #{e.message}.")
```

Comment:
The idiom to output full exception diagnostics is this:

`"Populator failed to connect a connection: #{e.class}: #{e.message}"`